### PR TITLE
Puppet manifests validation

### PR DESCRIPTION
**Pull request by vsellier**
Add a new image for validating puppet manifests

Everything is embedded, no additional parameters are needed. 

@mgreau tell me if it's ok, I don't know if it can work like this with jenkins

### CODE REVIEW

**Comment by fdrouet** [comment]
Some doc on your new image usage is missing in the README.md file

File: puppet/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ENV EXO_CI_LOG_DIR    ${EXO_CI_BASE}/logs
+ENV EXO_CI_TMP_DIR    ${EXO_CI_BASE}/tmp
+
+# Create user and group with specific ids
+RUN useradd --create-home --user-group -u 13000 --shell /bin/bash ${EXO_CI_USER}
+# giving all rights to eXo CI user
+RUN echo "ciagent  ALL = NOPASSWD: ALL" > /etc/sudoers.d/ciagent && chmod 440 /etc/sudoers.d/ciagent
+
+# Install puppet tools
+RUN apt-get update && apt-get install -y puppet && gem install r10k -v 2.4.3
```
Comment (fdrouet) [inline puppet/Dockerfile]: The puppet version installed is too old 3.4.x compared to the one we use.
Add these lines before puppet package install : 

```
# Install puppetlab repository
RUN gpg --keyserver keyserver.ubuntu.com --recv-key 4BD6EC30  \
  && gpg -a --export 4BD6EC30 | apt-key add -  \
  && echo "# Sources for recent versions of puppet" > /etc/apt/sources.list.d/puppetlab.list  \
  && echo "deb http://apt.puppetlabs.com $(lsb_release -c -s) main dependencies" >> /etc/apt/sources.list.d/puppetlab.list  \
  && echo "deb-src http://apt.puppetlabs.com $(lsb_release -c -s) main dependencies" >> /etc/apt/sources.list.d/puppetlab.list
```

**Comment by vsellier** [comment]
I have updated the image to fix the different issues :
- puppet is now on version 3.8.7
- r10k is now on version 2.0.3
- modules update is now optional and can be activated passing a -u option
- add doc
- install git from standard repo

File: puppet/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
+
+# eXo CI User
+ENV EXO_CI_USER ciagent
+ENV EXO_GROUP ciagent
+ENV HOME /home/${EXO_CI_USER}
+
+# Required directories for ciagent
+ENV EXO_CI_BASE       /opt/ciagent
```
Comment (mgreau) [inline puppet/Dockerfile]: We should update all CI dockerfiles with /srv/ciagent in order to be consistent with all Jenkins agents

**Comment by mgreau** [comment]
LGTM

**Pull request closed without merging**